src/libkomb.py

```diff
@@ -0,0 +1,473 @@
+from mapcss import MapCSS, Condition
+import json
+import sys
+import mapcss.webcolors
+from optparse import OptionParser
+
+whatever_to_hex = mapcss.webcolors.webcolors.whatever_to_hex
+
+def to_mapbox_condition(condition):
+    t = condition.type
+    params = condition.params
+
+    # if params[0] == "::class":
+    if params[0][:2] == "::":
+        return True
+
+    if t == "eq":
+        return ["==", ["get", params[0]], params[1]]
+    if t == "ne":
+        return ["!=", ["get", params[0]], params[1]]
+    if t == "regex":
+        return True
+    if t == "true":
+        return ["==", ["get", params[0]], "yes"]
+    if t == "untrue":
+        return ["==", ["get", params[0]], "no"]
+    if t == "set":
+        return ["to-boolean", ["get", params[0]]]
+    if t == "unset":
+        return ["!", ["to-boolean", ["get", params[0]]]]
+    if t == "<":
+        return ["<", ["to-number", ["get", params[0]]], float(params[1])]
+    if t == "<=":
+        return ["<=", ["to-number", ["get", params[0]]], float(params[1])]
+    if t == ">":
+        return [">", ["to-number", ["get", params[0]]], float(params[1])]
+    if t == ">=":
+        return [">=", ["to-number", ["get", params[0]]], float(params[1])]
+
+    return True
+
+
+def to_mapbox_expression(values_by_zoom):
+    values_by_zoom = values_by_zoom.items()
+    values_by_zoom = sorted(values_by_zoom, key=lambda k: k[0])
+    j = 0
+    for i in range(1, len(values_by_zoom)):
+        if values_by_zoom[i][1] != values_by_zoom[i - 1][1]:
+            j += 1
+            values_by_zoom[j] = values_by_zoom[i]
+    values_by_zoom = values_by_zoom[0 : j + 1]
+
+    if len(values_by_zoom) == 1:
+        return values_by_zoom[0][1]
+
+    expression = ["step", ["zoom"], values_by_zoom[0][1]]
+    for (zoom, value) in values_by_zoom:
+        expression.append(zoom)
+        expression.append(value)
+
+    return expression
+
+
+mapbox_linecaps = {"none": "butt", "butt": "butt", "round": "round", "square": "square"}
+
+def build_kepler_hints(conditions):
+    hints = []
+    for condition in conditions:
+        if condition.params[0] == "boundary":
+            hints.append("border")
+        if condition.params[0] == "highway":
+            hints.append("road")
+        if condition.params[0] == "place":
```

Comment:
this needs to check that there is text: xxx in the style.

komap has a mode in -r mapnik where it can separately output styles for background (reusable) and texts (localized).